core/src/main/java/com/datastax/oss/driver/internal/core/type/codec/registry/CachingCodecRegistry.java

```diff
@@ -413,4 +423,14 @@ protected boolean matches(TypeCodec<?> codec, GenericType<?> javaType, boolean i
     TypeCodec<DeclaredT> result = (TypeCodec<DeclaredT>) codec;
     return result;
   }
+
+  // These are mock types that are used as placeholders when we try to find a codec for an empty
+  // Java collection instance. All empty collections are serialized in the same way, so any element
+  // type will do:
+  private static final GenericType<List<Boolean>> JAVA_TYPE_FOR_EMPTY_LISTS =
+      GenericType.listOf(Boolean.class);
+  private static final GenericType<Set<Boolean>> JAVA_TYPE_FOR_EMPTY_SETS =
+      GenericType.setOf(Boolean.class);
+  private static final GenericType<Map<Boolean, Boolean>> JAVA_TYPE_FOR_EMPTY_MAPS =
```

Comment:
If the type is not important why we don't pass Object?
Or maybe even better we could create marker `class Any` and use it here for a readability

Reply:
Because the registry will still look for a codec for the element type, and there is no `Object` codec. We pick one of the primitive CQL types.